rdm4lib/src/vertex.rs

```diff
@@ -511,11 +595,49 @@ impl FromStr for TargetVertexFormat {
             "P4h_N4b_G4b_B4b_T2h" => Ok(TargetVertexFormat::P4h_N4b_G4b_B4b_T2h),
             "P4h_N4b_G4b_B4b_T2h_I4b" => Ok(TargetVertexFormat::P4h_N4b_G4b_B4b_T2h_I4b),
             "P4h_N4b_G4b_B4b_T2h_I4b_W4b" => Ok(TargetVertexFormat::P4h_N4b_G4b_B4b_T2h_I4b_W4b),
+            "P3f_N3f_G3f_B3f_T2f_C4b" => Ok(TargetVertexFormat::P3f_N3f_G3f_B3f_T2f_C4b),
             _ => Err(format!("Invalid value for VertexFormat: {}", input)),
         }
     }
 }
 
+pub trait VertexFormatProperties {
+    fn has_weights(vertex_format: &TargetVertexFormat) -> bool;
+    fn has_joints(vertex_format: &TargetVertexFormat) -> bool;
+    fn has_color(vertex_format: &TargetVertexFormat) -> bool;
+
+    //this is for later when we want to implementmultiple indices and weights 
+    fn weight_count() -> u8; 
+    fn joint_count() -> u8;
+    fn color_count() -> u8;  
```

Comment:
I'd prefer to have these named joint_component_count, etc. Otherwise, I'd confuse this with the total to number of joints, e.g. 16 for P4h_N4b_G4b_B4b_T2h_I4b_I4b_I4b_I4b_W4b_W4b_W4b_W4b. But as it stands, this should return 4.

Also, change the comment to something like this:
TODO: gltf2rdm make it possible to use more than one of these components (sets of 4 values).

Reply:
also reasonable, will also change that. I have already added support for multiple color sets. The only thing in the way is that blender gltf export cannot export multiple vertex colors... See https://github.com/KhronosGroup/glTF-Blender-IO/issues/2091